ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetProtocolSpecs.java

```diff
@@ -690,17 +686,19 @@ static ProtocolSpecBuilder shanghaiDefinition(
                     londonFeeMarket,
                     CoinbaseFeePriceCalculator.eip1559()))
         // Contract creation rules for EIP-3860 Limit and meter intitcode
-        .contractCreationProcessorBuilder(
-            (gasCalculator, evm) ->
-                new ContractCreationProcessor(
+        .transactionValidatorBuilder(
+            gasCalculator ->
+                new MainnetTransactionValidator(
                     gasCalculator,
-                    evm,
+                    londonFeeMarket,
                     true,
-                    List.of(
-                        MaxCodeSizeRule.of(contractSizeLimit),
-                        CachedInvalidCodeRule.of(EvmSpecVersion.SHANGHAI)),
-                    1,
-                    SPURIOUS_DRAGON_FORCE_DELETE_WHEN_EMPTY_ADDRESSES))
+                    chainId,
+                    Set.of(
+                        TransactionType.FRONTIER,
+                        TransactionType.ACCESS_LIST,
+                        TransactionType.EIP1559),
+                    quorumCompatibilityMode,
+                    SHANGHAI_INIT_CODE_SIZE_LIMIT))
```

Comment:
@shemnon I just spotted this...should it be
configContractSizeLimit.orElse(SHANGHAI_INIT_CODE_SIZE_LIMIT)?

Reply:
No.  We are limiting initcocde, Which is twice the contract size limit.  initcode and contract code have two different limits.